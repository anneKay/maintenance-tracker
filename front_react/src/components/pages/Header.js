import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from '../../img/mt-logo.png';

const Header = ({
  id, headerName, pathOne, pathThree, navThree, pathTwo, navOne, navTwo, className, pathname,
}) => (
  <header id={id} className={className}>
    <div>
      {(pathname === '/' || pathname === '/login' || pathname === '/signup') ? (
        <NavLink to="/" className="logo logo-primary">
          <img alt="" className="logo" src={logo} />
        </NavLink>
      ) : (pathname === '/profile') ? (
        <Link to="/profile" className="logo">

          <h3>
            Welcome &nbsp;&nbsp;
            {headerName}
          </h3>
        </Link>
      ) : <Link to="/profile"><h3>Dashboard</h3></Link>}
      <nav className="nav nav-primary">
        {pathOne ? <NavLink to={pathOne} activeClassName="is-active" className="btn btn-nav" exact>{navOne}</NavLink> : <input className="search" id="search" type="text" placeholder="Search by status" name="search" /> }
        <NavLink to={pathTwo} activeClassName="is-active" className="btn btn-nav">{navTwo}</NavLink>
        <NavLink to={pathThree} activeClassName="is-active" className="btn btn-nav">{navThree}</NavLink>

      </nav>
    </div>
  </header>

);

Header.defaultProps = {
  id: '',
  pathname: '',
  headerName: '',
};

Header.propTypes = {
  id: PropTypes.string,
  headerName: PropTypes.string,
  pathOne: PropTypes.string.isRequired,
  pathTwo: PropTypes.string.isRequired,
  navOne: PropTypes.string.isRequired,
  navTwo: PropTypes.string.isRequired,
  navThree: PropTypes.string.isRequired,
  pathThree: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  pathname: PropTypes.string,
};
export default Header;
