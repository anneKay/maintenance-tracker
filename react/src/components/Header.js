import React from 'react';

const Header = (props) => (
   <div>
        <h1>This is a header </h1>
        <p>{props.title}</p>
    </div>
)
export default Header