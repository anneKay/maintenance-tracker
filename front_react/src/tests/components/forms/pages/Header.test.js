import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../../components/pages/Header';


describe(' Header Form component rendering', () => {
  const props = {
    id: 'headerId',
    headerName: 'Nwanna',
    pathOne: '/',
    pathThree: '/login',
    navThree: 'Home',
    pathTwo: '/signup',
    navOne: 'Logout',
    navTwo: 'Report Issue',
    className: 'Header Primary',
  };
  const wrapper = shallow(<Header {...props} />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
