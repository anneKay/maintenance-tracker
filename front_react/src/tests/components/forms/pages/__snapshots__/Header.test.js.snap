// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Header Form component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    className="Header Primary"
    headerName="Nwanna"
    id="headerId"
    navOne="Logout"
    navThree="Home"
    navTwo="Report Issue"
    pathOne="/"
    pathThree="/login"
    pathTwo="/signup"
    pathname=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <Link
          replace={false}
          to="/profile"
        >
          <h3>
            Dashboard
          </h3>
        </Link>
        <nav
          className="nav nav-primary"
        >
          <NavLink
            activeClassName="is-active"
            aria-current="page"
            className="btn btn-nav"
            exact={true}
            to="/"
          >
            Logout
          </NavLink>
          <NavLink
            activeClassName="is-active"
            aria-current="page"
            className="btn btn-nav"
            to="/signup"
          >
            Report Issue
          </NavLink>
          <NavLink
            activeClassName="is-active"
            aria-current="page"
            className="btn btn-nav"
            to="/login"
          >
            Home
          </NavLink>
        </nav>
      </div>,
      "className": "Header Primary",
      "id": "headerId",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/profile"
          >
            <h3>
              Dashboard
            </h3>
          </Link>,
          <nav
            className="nav nav-primary"
          >
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              exact={true}
              to="/"
            >
              Logout
            </NavLink>
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              to="/signup"
            >
              Report Issue
            </NavLink>
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              to="/login"
            >
              Home
            </NavLink>
          </nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              Dashboard
            </h3>,
            "replace": false,
            "to": "/profile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dashboard",
            },
            "ref": null,
            "rendered": "Dashboard",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                exact={true}
                to="/"
              >
                Logout
              </NavLink>,
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                to="/signup"
              >
                Report Issue
              </NavLink>,
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                to="/login"
              >
                Home
              </NavLink>,
            ],
            "className": "nav nav-primary",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Logout",
                "className": "btn btn-nav",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Logout",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Report Issue",
                "className": "btn btn-nav",
                "to": "/signup",
              },
              "ref": null,
              "rendered": "Report Issue",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Home",
                "className": "btn btn-nav",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <Link
            replace={false}
            to="/profile"
          >
            <h3>
              Dashboard
            </h3>
          </Link>
          <nav
            className="nav nav-primary"
          >
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              exact={true}
              to="/"
            >
              Logout
            </NavLink>
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              to="/signup"
            >
              Report Issue
            </NavLink>
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              className="btn btn-nav"
              to="/login"
            >
              Home
            </NavLink>
          </nav>
        </div>,
        "className": "Header Primary",
        "id": "headerId",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/profile"
            >
              <h3>
                Dashboard
              </h3>
            </Link>,
            <nav
              className="nav nav-primary"
            >
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                exact={true}
                to="/"
              >
                Logout
              </NavLink>
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                to="/signup"
              >
                Report Issue
              </NavLink>
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                className="btn btn-nav"
                to="/login"
              >
                Home
              </NavLink>
            </nav>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h3>
                Dashboard
              </h3>,
              "replace": false,
              "to": "/profile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": "h3",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="is-active"
                  aria-current="page"
                  className="btn btn-nav"
                  exact={true}
                  to="/"
                >
                  Logout
                </NavLink>,
                <NavLink
                  activeClassName="is-active"
                  aria-current="page"
                  className="btn btn-nav"
                  to="/signup"
                >
                  Report Issue
                </NavLink>,
                <NavLink
                  activeClassName="is-active"
                  aria-current="page"
                  className="btn btn-nav"
                  to="/login"
                >
                  Home
                </NavLink>,
              ],
              "className": "nav nav-primary",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "aria-current": "page",
                  "children": "Logout",
                  "className": "btn btn-nav",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "aria-current": "page",
                  "children": "Report Issue",
                  "className": "btn btn-nav",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Report Issue",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "aria-current": "page",
                  "children": "Home",
                  "className": "btn btn-nav",
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
            ],
            "type": "nav",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
